### PR TITLE
Add is_integer to SymFloat

### LINKED ISSUE #114676: 'SymFloat' object has no attribute 'is_integer'
### 🐛 Describe the bug
```
import torch

@torch.compile(fullgraph=True, dynamic=True, backend="inductor")
def test(x, mul, dim=-1):
    size = x.size(dim)
    m = size/mul
    if m.is_integer():
        return m
    return size

t = torch.randn((3, 6, 4, 2), requires_grad = False)
test(t, 2)
```

Error logs:

  File "../lib/python3.8/site-packages/torch/_dynamo/eval_frame.py", line 328, in _fn
    return fn(*args, **kwargs)
  File "../lib/python3.8/site-packages/torch/_dynamo/eval_frame.py", line 490, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state)
  File "../lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 133, in _fn
    return fn(*args, **kwargs)
  File "../lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 389, in _convert_frame_assert
    return _compile(
  File "../lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 569, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "../lib/python3.8/site-packages/torch/_dynamo/utils.py", line 189, in time_wrapper
    r = func(*args, **kwargs)
  File "../lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 491, in compile_inner
    out_code = transform_code_object(code, transform)
  File "../lib/python3.8/site-packages/torch/_dynamo/bytecode_transformation.py", line 1028, in transform_code_object
    transformations(instructions, code_options)
  File "../lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 458, in transform
    tracer.run()
  File "../lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 2074, in run
    super().run()
  File "../lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 724, in run
    and self.step()
  File "../lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 688, in step
    getattr(self, inst.opname)(inst)
  File "../lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 392, in wrapper
    return inner_fn(self, inst)
  File "../lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 1115, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "../lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 562, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "../lib/python3.8/site-packages/torch/_dynamo/variables/misc.py", line 594, in call_function
    return self.obj.call_method(tx, self.name, args, kwargs).add_options(self)
  File "../lib/python3.8/site-packages/torch/_dynamo/variables/tensor.py", line 716, in call_method
    return wrap_fx_proxy(
  File "../lib/python3.8/site-packages/torch/_dynamo/variables/builder.py", line 1187, in wrap_fx_proxy
    return wrap_fx_proxy_cls(
  File "../lib/python3.8/site-packages/torch/_dynamo/variables/builder.py", line 1274, in wrap_fx_proxy_cls
    example_value = get_fake_value(proxy.node, tx)
  File "../lib/python3.8/site-packages/torch/_dynamo/utils.py", line 1376, in get_fake_value
    raise TorchRuntimeError(str(e)).with_traceback(e.__traceback__) from None
  File "../lib/python3.8/site-packages/torch/_dynamo/utils.py", line 1337, in get_fake_value
    return wrap_fake_exception(
  File "../lib/python3.8/site-packages/torch/_dynamo/utils.py", line 916, in wrap_fake_exception
    return fn()
  File "../lib/python3.8/site-packages/torch/_dynamo/utils.py", line 1338, in <lambda>
    lambda: run_node(tx.output, node, args, kwargs, nnmodule)
  File "../lib/python3.8/site-packages/torch/_dynamo/utils.py", line 1410, in run_node
    raise RuntimeError(fn_str + str(e)).with_traceback(e.__traceback__) from e
  File "../lib/python3.8/site-packages/torch/_dynamo/utils.py", line 1399, in run_node
    return getattr(args[0], node.target)(*args[1:], **kwargs)
torch._dynamo.exc.TorchRuntimeError: Failed running call_method is_integer(*(s3/s4,), **{}):
'SymFloat' object has no attribute 'is_integer'

### Versions

Collecting environment information...
PyTorch version: 2.1.0a0+gite999726
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.31

Python version: 3.8.18 (default, Sep 11 2023, 13:40:15)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.4.0-163-generic-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      43 bits physical, 48 bits virtual
CPU(s):                             12
On-line CPU(s) list:                0-11
Thread(s) per core:                 1
Core(s) per socket:                 1
Socket(s):                          12
NUMA node(s):                       1
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              85
Model name:                         Intel(R) Xeon(R) Gold 5220R CPU @ 2.20GHz
Stepping:                           0
CPU MHz:                            2194.843
BogoMIPS:                           4389.68
Virtualization:                     VT-x
Hypervisor vendor:                  VMware
Virtualization type:                full
L1d cache:                          384 KiB
L1i cache:                          384 KiB
L2 cache:                           12 MiB
L3 cache:                           429 MiB
NUMA node0 CPU(s):                  0-11
Vulnerability Gather data sampling: Unknown: Dependent on hypervisor status
Vulnerability Itlb multihit:        KVM: Mitigation: Split huge pages
Vulnerability L1tf:                 Mitigation; PTE Inversion; VMX flush not necessary, SMT disabled
Vulnerability Mds:                  Mitigation; Clear CPU buffers; SMT Host state unknown
Vulnerability Meltdown:             Mitigation; PTI
Vulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT Host state unknown
Vulnerability Retbleed:             Mitigation; IBRS
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; IBRS, IBPB conditional, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon nopl xtopology tsc_reliable nonstop_tsc cpuid pni pclmulqdq vmx ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 invpcid avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xsaves arat pku ospke md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] torch==2.1.0a0+gite999726
[conda] numpy                     1.24.4                   pypi_0    pypi

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### PR DESCRIPTION
**Pull request by isuruf**
Fixes #114676

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/fx/experimental/sym_node.py [hunk 1]
```diff
@@ -442,13 +446,23 @@ def is_constant(self):
     "sym_not",
 }
 
+# Unary methods that are not magids that are not magicc
```
Comment (jon-chuang) [inline torch/fx/experimental/sym_node.py:449]: typo

File: torch/fx/experimental/sym_node.py [hunk 2]
```diff
@@ -886,6 +908,8 @@ def unary_magic_impl(self):
         pytype: Type
         if method in always_int_magic_methods:
             pytype = int
+        elif method in always_bool_magic_methods:
```
[the remark this answers is not in the document]
Comment (isuruf) [reply torch/fx/experimental/sym_node.py:911]: Previously all the unary `always_bool_magic_methods` has `bool` as it's input and there the `else:` branch was taken and was correct.

File: torch/fx/experimental/sym_node.py [hunk 3]
```diff
@@ -1109,6 +1133,8 @@ def rbinary_magic_impl(self, other):
 
     if method in unary_magic_methods:
         setattr(user_type, f"__{method}__", unary_magic_impl)
+    elif method in unary_nonmagic_methods:
+        setattr(user_type, method, unary_magic_impl)
```
Comment (peterbell10) [inline torch/fx/experimental/sym_node.py:1137]: These methods need docs, you could do this by writing a stub in `SymFloat` and use `functolls.wraps` here to take the docs from the stub.

**Review by peterbell10 — APPROVED** [review]
API documentation test is failing, otherwise LGTM.

**Pull request closed without merging**